Neos.ContentRepository.Core/Classes/Projection/ContentGraph/Filter/SearchTerm/SearchTerm.php

```diff
@@ -12,7 +12,7 @@
 
 declare(strict_types=1);
 
-namespace Neos\ContentRepository\Core\Projection\ContentGraph;
+namespace Neos\ContentRepository\Core\Projection\ContentGraph\Filter\SearchTerm;
```

Comment:
I think, we had it in the main ContentGraph folder by intention because it is likely to be reused for other parts.
Can be discussed, but IMO we should not sneak in such changes in unrelated PRs as it makes the review needlessly harder and could confuse debugging

Reply:
Yep, I would also let it stay where it was.

Reply:
With the rest of the changes I came to understand why you moved it

Reply:
I would almost argue the SearchTermMatcher doesn't belong in Projection/ContentGraph/Filters and that strengthens the argument the SearchTerm should be generic as well. Yes the SearchTermMatcher applies to contentgraph projection nodes, BUT I tihnk it's clear that everything else in filters is used in conjunction with a ContentGraph or Subgraph query (query in the loosest term not necessarily DB query), while the SearchTermMatcher applies to a set of nodes that have already been assembled. I think that is its own kind of operation.

Reply:
i moved it to align it to the `PropertyValueCriteriaMatcher` it resides as well in the `PropertyValue` namespace and it was only logical to me that the same should apply for `SearchTerm` things. Id be fine with that non-ideal namespace for now as the `SearchTermMatcher` is marked as `@internal` for now and we can later decide on a better location for this and the `PropertyValueCriteriaMatcher`. Or is there a better suggestion out there? Probably with the introduction of the inmemory content graph well have a better namespace at hand.